driver-async/src/main/com/mongodb/async/client/OperationIterable.java

```diff
@@ -117,6 +117,12 @@ public void onResult(final List<T> results, final Throwable t) {
         return new MappingIterable<T, U>(this, mapper);
     }
 
+    @Override
+    public OperationIterable<T> batchSize(final int batchSize) {
```

Comment:
That example is fine as `find()` uses the FindFluent that supports batchSize() and its applied to the delegate before the forEach().  

However any OperationIterables don't support batchSize but throwing an unsupported error defeats the purpose of promoting batchSize to the iterable.